### PR TITLE
feat(configuration): add os-specific paths for dprint binary

### PR DESCRIPTION
**Pull request by Shqrp**
I found myself working on three different OSes and changing the path with Settings Sync became an hassle, so I'm proposing platform-specific paths for every OS.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,16 +32,39 @@
       "type": "object",
       "title": "dprint",
       "properties": {
-        "dprint.path": {
```
Comment (dsherret) [inline package.json:35]: We should still keep `"dprint.path"`.

File: src/FolderService.ts [hunk 2]
```diff
@@ -165,7 +165,20 @@ export class FolderService implements vscode.DocumentFormattingEditProvider {
       return path == null ? undefined : shellExpand(path);
 
       function getRawPath() {
-        const path = config.get("path");
+        let path: string | undefined;
+        console.log(config.get("linuxPath"));
```
Comment (dsherret) [inline src/FolderService.ts:169]:
```suggestion
```

**Review by dsherret — COMMENTED** [review]
Can you explain your use case a bit more? Is this for your global user settings or project specific (workspace) settings?

> I found myself working on three different OSes and changing the path with Settings Sync became an hassle

What about using `"settingsSync.ignoredSettings": [ "dprint.path" ]`? https://code.visualstudio.com/docs/editor/settings-sync#_configuring-synced-data

**Comment by Shqrp** [comment]
Ignoring `dprint.path` in Settings Sync is definitely a valid alternative, but I can see how implementing this directly can make the dprint user experience much more seamless: considering that most users would install dprint in the default path, Settings Sync with OS-specific paths can provide a much less friction experience.

**Pull request closed without merging**